### PR TITLE
chore: add supported versions to docs

### LINKED ISSUE #80: Add release interoperabilit
### Code of Conduct

- [X] I have read and agree to the project's Code of Conduct.

### Provider Version

All

### Motivation

We've neglected providing the version interoperability detail for the provider in the releases and documentation.

In addition, the SDK is based on v4.5.x so there may be some incompatibility for v5.x.

### Suggestion

- Update `README.md` to reference the documentation on the Terraform Registry.
- Update `/docs/index.md` with a  table that provides interoperability information.
- Update `CHANGELOG.md` with:
  - Add reference the documentation on the Terraform Registry.
  - For interoperability changes:
     - Adds support for VMware Cloud Foundation _x_.
     - BREAKING CHANGE: Removes support for VMware Cloud Foundation _x_.
 
   This content will be copied to the GitHub release details.

### PR DESCRIPTION
**Pull request by stoyanzhelyazkov**
**Summary of Pull Request**

Adding the supported VCF version to the docs

**Type of Pull Request**

- [ ] This is a bug fix.
- [ ] This is an enhancement or feature.
- [ ] This is a code style/formatting update.
- [X] This is a documentation update.
- [ ] This is a refactoring update.
- [X] This is a chore update
- [ ] This is something else.
      Please describe:

**Related to Existing Issues**

Closes #80

**Test and Documentation Coverage**

For bug fixes or features:

- [ ] Tests have been completed.
- [X] Documentation has been added/updated.

**Breaking Changes?**

- [ ] Yes, there are breaking changes.
- [X] No, there are no breaking changes.

### CODE REVIEW

**Review by dimitarproynov — DISMISSED** [review]

**Comment by stoyanzhelyazkov** [comment]
Hey @tenthirtyam I know it's not exactly what you asked for in https://github.com/vmware/terraform-provider-vcf/issues/80

We don't really support anything past 4.5.2 and although most if not all resources work with 5.X
we can't declare that officially just yet.

I did add some text to the docs that says these things explicitly
Let me know if this works for you

File: docs/index.md [hunk 1]
```diff
@@ -8,7 +8,11 @@ description: |-
 
 # Terraform Provider for VMware Cloud Foundation
 
+This provider works with versions of VMware Cloud Foundation up to and including 4.5.2
```
Comment (tenthirtyam) [inline docs/index.md:11]: Perhaps 4.4 to 4.5.2?

File: docs/index.md [hunk 2]
```diff
@@ -8,7 +8,11 @@ description: |-
...
 
+See the [release notes](https://docs.vmware.com/en/VMware-Cloud-Foundation/4.5.2/rn/vmware-cloud-foundation-452-release-notes/index.html) for the individual build numbers
+
+Interoperability with later versions of VMware Cloud Foundation is possible but not guaranteed
```
Comment (tenthirtyam) [inline docs/index.md:15]:
```suggestion
Interoperability with later versions of VMware Cloud Foundation is not assured.
```

File: docs/index.md [hunk 3]
```diff
@@ -8,7 +8,11 @@ description: |-
 
 # Terraform Provider for VMware Cloud Foundation
 
+This provider works with versions of VMware Cloud Foundation up to and including 4.5.2
 
+See the [release notes](https://docs.vmware.com/en/VMware-Cloud-Foundation/4.5.2/rn/vmware-cloud-foundation-452-release-notes/index.html) for the individual build numbers
```
Comment (tenthirtyam) [inline docs/index.md:13]:
```suggestion
See the [release notes](https://docs.vmware.com/en/VMware-Cloud-Foundation/4.5.2/rn/vmware-cloud-foundation-452-release-notes/index.html) for the individual build numbers.
```

**Review by tenthirtyam — CHANGES_REQUESTED** [review]

File: docs/index.md (hunk 1 above)
Comment (stoyanzhelyazkov) [reply docs/index.md]: Changed to `This provider works with VMware Cloud Foundation 4.4.0 to 4.5.2`

**Review by tenthirtyam — APPROVED** [review]
LGTM!

**Pull request merged**